packages/compass-indexes/src/components/create-index-form/index-flow-section.tsx

```diff
@@ -107,6 +173,26 @@ const IndexFlowSection = ({
     {}
   );
 
+  const coveredQueriesArr = fields.map((field, index) => {
+    return { [field.name]: index + 1 };
+  });
+
+  const [coveredQueriesExamples, setCoveredQueriesExamples] =
+    useState<JSX.Element>(<></>);
+  const [optimalQueriesExamples, setOptimalQueriesExamples] = useState<
+    string | JSX.Element
+  >('');
+  const [showCoveredQueries, setShowCoveredQueries] = useState(false);
+
+  const onCoveredQueriesButtonClick = () => {
```

Comment:
We should use `useCallback` here so that we don't re-create this method on every render. https://react.dev/reference/react/useCallback

Reply:
Looks like we could also be creating the `coveredQueriesArr` in this function instead of every render as well.